src/crypto/symmkey.cpp

```diff
@@ -14,49 +15,55 @@ namespace crypto
 {
   KeyAesGcm::KeyAesGcm(CBuffer rawKey)
   {
-    ctx = new mbedtls_gcm_context;
-    mbedtls_gcm_init(reinterpret_cast<mbedtls_gcm_context*>(ctx));
-
-    size_t n_bits;
-    const auto n = static_cast<unsigned int>(rawKey.rawSize() * 8);
-    if (n >= 256)
-    {
-      n_bits = 256;
-    }
-    else if (n >= 192)
-    {
-      n_bits = 192;
-    }
-    else if (n >= 128)
-    {
-      n_bits = 128;
-    }
-    else
+    auto thread_count = enclave::ThreadMessaging::max_num_threads;
```

Comment:
`max_num_threads` should be constexpr, then `ctxs` can be `std::array<mbedtls_gcm_context, enclave::ThreadMessaging::max_num_threads>` (unless there's some reason I'm missing this actually needs to be `void*`?) and we don't need to resize.